test/test_optim.py

```diff
@@ -1039,7 +1039,8 @@ def _test_lr_is_constant_for_constant_epoch(self, scheduler):
 
         for _ in range(10):
             scheduler.optimizer.step()
-            scheduler.step(2)
+            with warnings.catch_warnings(record=True) as w:
```

Comment:
Could you add a comment just before the catcher here to mention that we are hiding the warning due to the scheduler being used without an optimizer on purpose here.
Also to be extra sure that we don't hide anything else, you can add a `self.assertEqual(len(w), 1)` after the block to make sure we didn't catch any other unexpected warning.

Reply:
Thanks for the suggestion -- will do!

Reply:
Correct me if I am wrong, but the way I read this test:
we are using an optimizer, the reason we get a warning is because we are passing in the `epoch` parameter -- to test that `step(2)` is idempotent.

Reply:
For this one looks like so.